File: IOS/LuaViewSDK/Lua/Local/UI/LayoutFactory.m
//
//  LayoutFactory.m
//  LuaViewSDK
//
//  Created by 王金东 on 17/2/9.
//  Copyright © 2017年 王金东. All rights reserved.
//

#import "LayoutFactory.h"
#import "HorizontalLayout.h"
#import "VerticalLayout.h"
#import "RelativeLayout.h"
#import "HorizontalEqualWidthLayout.h"
#import "VerticalEqualHeightLayout.h"

@implementation LayoutFactory

+ (id<LuaLayout>)getFactory:(LAYOUT_STYLE)layout {
    if (layout == Horizontal_layout) {
        return [[HorizontalLayout alloc] init];
    }else if(layout == Vertical_layout){
        return [[VerticalLayout alloc] init];
    }else if(layout == Relative_layout){
        return [[RelativeLayout alloc] init];
    }else if(layout == Vertical_equal_height_layout){
        return [[VerticalEqualHeightLayout alloc] init];
    }else if(layout == Horizontal_equal_width_layout){
        return [[HorizontalEqualWidthLayout alloc] init];
    }
    return [[RelativeLayout alloc] init];
}
@end

File: Android/LuaViewSDK/app/src/main/java/com/core/lua/local/ui/LayoutFactory.java
package com.core.lua.local.ui;

/**
 * Created by JD on 2017/2/9.
 */
public class LayoutFactory {
    public static LuaLayout getFactory(LuaViewGroup.Layout layout_type){
        if (layout_type == LuaViewGroup.Layout.HORIZONTAL_LAYOUT){
              return new HorizontalLayout();
        }else  if (layout_type == LuaViewGroup.Layout.RELATIVE_LAYOUT){
            return new RelativeLayout();
        }else if(layout_type == LuaViewGroup.Layout.VERTICAL_LAYOUT){
           return new VerticalLayout();
        }else if(layout_type == LuaViewGroup.Layout.HORIZONTAL_EQUAL_WIDTH_LAYOUT){
           return new HorizontalEqualWidthLayout();
        }else if(layout_type == LuaViewGroup.Layout.VERTICAL_EQUAL_HEIGHT_LAYOUT){
            return new VerticalEqualHeightLayout();
        }else {
            return new RelativeLayout();
        }
    }
}
